### PR TITLE
Set credentials for commiting to VCS with

### PR DESCRIPTION
**Pull request by joh6nn**

### CODE REVIEW

File: tasks/bzr.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+- name: Set etckeeper bzr whoami
+  become: yes
```
Comment (ypid) [inline tasks/bzr.yml]: The playbook is already expected to be run with `become: True`. No need to repeat for all tasks. (Playbook can be found here: https://github.com/debops-contrib/debops-contrib-playbooks/blob/master/service/etckeeper.yml)

File: tasks/git.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+ - name: "Set etckeeper git user.name"
+   sudo: yes
```
Comment (ypid) [inline tasks/git.yml]: `sudo` has been deprecated. You can remove it. See above comment.

File: tasks/bzr.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+  command: "etckeeper vcs whoami '{{etckeeper_vcs_username}} <{{etckeeper_vcs_email}}>'"
+  tags: [ 'etckeeper_creds' ]
```
Comment (ypid) [inline tasks/bzr.yml]: Tags currently don’t work in include files. http://www.ansible.com/blog/ansible-2.0-launch Also the DebOps project uses a specific format for tags. I would suggest `role::etckeeper:commit_options`.

File: defaults/main.yml [hunk 4]
```diff
@@ -46,6 +46,15 @@ etckeeper_lowlevel_package_manager: |
 #
 etckeeper_vcs: 'git'
 
+# .. envvar:: etckeeper_vcs_username
+#
+# username to use when making VCS commits
+etckeeper_vcs_username: "{{ansible_hostname}}"
```
Comment (ypid) [inline defaults/main.yml]: Just a convention thing. Can you change that to `etckeeper_vcs_username: '{{ ansible_hostname }}'` maybe? This makes it easier to copy paste stuff. When you need to quote inside Jinja, we use `"` for this for example: `etckeeper_vcs_username: '{{ ansible_hostname + "-staging" }}'`.

**Comment by ypid** [comment]
Thanks for the PR. As the include tasks mostly only consist of one task I would prefer to include them in the main file. Can you change this maybe? Note that any other VCS than git is not tested with this role. It is nice of course to support them but I am not sure if people are still using some of them so this will probably just be overhead :wink: I intended to wait until an issue or PR for a specify VCS is open before adding it.

File: defaults/main.yml [hunk 5]
```diff
@@ -46,6 +46,15 @@ etckeeper_lowlevel_package_manager: |
 #
 etckeeper_vcs: 'git'
 
+# .. envvar:: etckeeper_vcs_username
+#
+# username to use when making VCS commits
```
Comment (ypid) [inline defaults/main.yml]: Another small thing. Can you change the style to match the rest of the file. E.g. start a sentence with a upper case letter and end with a full stop? Sorry for the many comments :wink:

**Comment by joh6nn** [comment]
no worries on the many comments! this is my first contribution to a role, so I was pretty sure there'd be a bunch of consistency and style issues. gotta start somewhere though!

**Comment by ypid** [comment]
All right. Welcome to the DebOps project!

**Comment by joh6nn** [comment]
Thanks! Here's round 2, hopefully that looks a little cleaner

**Comment by ypid** [comment]
Merged. Thanks. You might want to check my [rework commit](https://github.com/debops-contrib/ansible-etckeeper/commit/613d65d8ff7dab3f9770062434c03cc164bb5e3d) to further improve your Ansible and DebOps foo. Most important rework was idempotents related.

**Comment by ypid** [comment]
The change from "username" to "user" was done for consistency with other roles.

**Pull request merged**